libnmstate/nm/port_vlan.py

```diff
@@ -0,0 +1,238 @@
+        return self._tag
+
+    @property
+    def enable_native_vlan(self):
+        return self._is_native_vlan
+
+    @property
+    def port_type(self):
+        return self._port_type
+
+    def generate(self):
+        """
+        Generate a list of NM.BridgeVlan objects from the encapsulated
+        PortVlanFilter data
+        """
+
+        port_vlan_config = []
+        if self.port_type == LB.Port.Vlan.TRUNK_TYPE:
+            for trunk_port in self.trunk_ports:
+                port_vlan_config.append(
+                    self._generate_vlan_trunk_port_config(trunk_port)
+                )
+            if self.enable_native_vlan and self.tag:
+                port_vlan_config.append(
+                    self._generate_vlan_access_port_config(self.tag)
+                )
+        elif self.port_type == LB.Port.Vlan.ACCESS_TYPE and self.tag:
+            port_vlan_config.append(
+                self._generate_vlan_access_port_config(self.tag)
+            )
+
+        return port_vlan_config
+
+    def to_dict(self):
+        """
+        Get the port vlan filtering configuration in dict format - e.g. in yaml
+        format:
+
+        - name: eth1
+          vlan:
+            type: trunk
+            trunk-tags:
+              - id: 101
+              - id-range:
+                  min: 200
+                  max: 4095
+            tag: 100
+            enable-native: true
+        """
+
+        port_vlan_state = {
+            LB.Port.Vlan.TYPE: self.port_type,
+            LB.Port.Vlan.TRUNK_TAGS: self.trunk_ports,
+        }
+        if self.tag:
+            port_vlan_state[LB.Port.Vlan.TAG] = self.tag
+        if self.port_type == LB.Port.Vlan.TRUNK_TYPE:
+            port_vlan_state[
+                LB.Port.Vlan.ENABLE_NATIVE
+            ] = self.enable_native_vlan
+        return port_vlan_state
+
+    @staticmethod
+    def build_from_port_vlans(port_vlans):
+        """
+        Factory method that instantiates a PortVlanFilter object from a
+        list of NM.BridgeVlan objects.
+        """
+
+        bridge_vlan_config = [
+            PortVlanFilter._get_vlan_info(port_vlan_filter_entry)
+            for port_vlan_filter_entry in port_vlans
+        ]
+
+        trunk_tags, tag, enable_native = PortVlanFilter._get_vlan_config(
+            bridge_vlan_config
+        )
+
+        return PortVlanFilter(trunk_tags, tag, enable_native)
+
+    @staticmethod
+    def get_vlan_tag_range(port_vlan):
+        """
+        Extract the vlan tags from the NM.BridgeVlan object.
+        A single NM.BridgeVlan object can have a range of vlan tags, or a
+        single one.
+
+        When a NM.BridgeVlan holds a single tag, the min_range, and max_range
+        returned will have the same vlan tag.
+
+        :return: min_range, max_range
+        """
+
+        port_vlan_tags = port_vlan.to_str()
+        split_port_vlan_tags = port_vlan_tags.split()
+
+        if split_port_vlan_tags and '-' in split_port_vlan_tags[0]:
```

Comment:
Why is `split_port_vlan_tags` part of the condition? In the else block, `split_port_vlan_tags[0]` is called which implies it does not evaluate to False or there will be an IndexError.

Reply:
Right.

Since these methods operate based on data that originates from libnm's `port_setting.props.vlans` , I can drop the 'split' condition altogether.